src/componenets/Calculator.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from 'react';
+import Big from 'big.js';
+import Button from './Button';
+import './calculator.css';
+import operate from '../logic/operate';
+
+function Calculator() {
+  const [previousOperation, setpreviousOperation] = useState('');
+  const [currentOperation, setcurrentOperation] = useState();
+  const [operation, setNewOperation] = useState();
+
+  const handleDigitClick = (digit) => {
+    if (digit === '.' && currentOperation.includes('.')) return;
+    if (currentOperation === '0' && digit !== '.') {
+      setcurrentOperation(digit);
+    } else {
+      setcurrentOperation(currentOperation + digit);
+    }
+  };
+
+  const handleOperationClick = (operation) => {
+    if (previousOperation === '') {
+      setpreviousOperation(currentOperation);
+      setcurrentOperation('0');
+      setNewOperation(operation);
+    } else {
+      const result = operate(previousOperation, currentOperation, operation);
+      setpreviousOperation(result);
+      setcurrentOperation('0');
+      setNewOperation(operation);
+    }
+  };
+
+  const handlePercentageClick = () => {
+    setcurrentOperation(Big(currentOperation).mod(100).toString());
+  };
```

Comment:
- [ ] Kindly fix the operation of `%`. In this project's scope, it means the modules sign not the percent sign.
### Hint
There is no need for a particular case to handle this operation as it's already handled in the provided files `calculate.js`, and `operate.js`.